syntax = "proto3";


service Test {
  rpc TimeConsumingMethod(TimeConsumingRequest) returns (TimeConsumingResponse) {}
  rpc GetShapes(SceneID) returns (stream Shape) {}
  rpc MachineStatus(stream MachineStatusRequest) returns (stream MachineStatusResponse) {}
}

message MachineStatusRequest {
  int32 machine_id = 1;
}

message MachineStatusResponse {
  int32 machine_id = 1;
  string status = 2;
  repeated string logs = 3;
}

message TimeConsumingRequest {
  string command = 1;
  bytes data = 2;
}

message TimeConsumingResponse {
  int32 status = 1;
  string message = 2;         
  bytes result = 3;
}


message Shape {
  string type = 1;           // e.g., "circle", "square"
  repeated float dimensions = 2; // e.g., radius for circle, side lengths for square
  string color = 3;          // e.g., "red", "blue"
}


message SceneID {
  int32 id=1;
}

